pkg/api/meta/multirestmapper.go

```diff
@@ -169,24 +170,29 @@ func (m MultiRESTMapper) RESTMapping(gk unversioned.GroupKind, versions ...strin
 	if len(allMappings) == 1 {
 		return allMappings[0], nil
 	}
+	if len(allMappings) > 1 {
+		return nil, fmt.Errorf("multiple matches found for %v in %v", gk, versions)
+	}
 	if len(errors) > 0 {
 		return nil, utilerrors.NewAggregate(errors)
 	}
-	if len(allMappings) == 0 {
-		return nil, fmt.Errorf("no match found for %v in %v", gk, versions)
-	}
-
-	return nil, fmt.Errorf("multiple matches found for %v in %v", gk, versions)
+	return nil, fmt.Errorf("no match found for %v in %v", gk, versions)
 }
 
 // AliasesForResource finds the first alias response for the provided mappers.
 func (m MultiRESTMapper) AliasesForResource(alias string) ([]string, bool) {
+	seenAliases := sets.NewString()
 	allAliases := []string{}
 	handled := false
 
 	for _, t := range m {
 		if currAliases, currOk := t.AliasesForResource(alias); currOk {
-			allAliases = append(allAliases, currAliases...)
+			for _, currAlias := range currAliases {
```

Comment:
> Shall we allow different RESTMappers to have same aliases? I think we should just make the resources that got aliased to be qualified with group

not sure about that... it makes sense for `hpa` to expand to `horizontalpodautoscalers` whether you're working in the `extensions` or `autoscaling` groups, but if the alias expanded to resource name AND group name in both mappers, then `hpa` would expand to `horizontalpodautoscalers.autoscaling,horizontalpodautoscalers.extensions`, which isn't what we want.

agree aliases need more thought, but I'd like to limit this PR to deduping the expanded list, which we know we want.